shell/src/main/java/org/apache/zeppelin/shell/ShellInterpreter.java

```diff
@@ -78,6 +106,10 @@ public void close() {
         }
       }
     }
+
+    if (shellOutputCheckExecutor != null) {
+      shellOutputCheckExecutor.shutdownNow();
```

Comment:
The utility class [`ExecutorUtil`](https://github.com/apache/zeppelin/blob/master/zeppelin-interpreter/src/main/java/org/apache/zeppelin/util/ExecutorUtil.java) provides a way to gently shut down this executor.

Reply:
Fixed